### PR TITLE
Be us3 add potential match field

### PR DESCRIPTION
**Pull request by MWMJohnson**
__X_ Wrote Tests _X__ Implemented __X_ Reviewed

Neccesary checkmarks:

  - [x] All Tests are Passing

  - [x] The code will run locally
  You will need to turn Sprockets/Railties in the config/application.rb file, for local to run.

Type of change

  - [x] New feature
  - [ ] Bug Fix

Implements/Fixes:
1. Add a query for FE to send a base 64 image string to BE, allowing BE to make a 3rd Party API call
and return that data in specifically requested with no over or under fetching all via GraphQL, as opposed to a REST architecture.

2. Still need to persist the data into the BE DB, to start adding Mushrooms. And Then add a refined called to a mushrooms field as opposed to the matches field which returns POROS objects instead of Model objects. 

3. Need to add a Mutation field for creating a user-mushroom, with a is_saved:boolean attribute.

    

Check the correct boxes

  - [x] This broke nothing
  - [ ] This broke some stuff
  - [ ] This broke everything

Testing Changes

   - [ ] No Tests have been changed
   - [x] Some Tests have been changed
   - [ ] All of the Tests have been changed(Please describe what in the world happened)

Checklist:

   - [ ] My code has no unused/commented out code
   - [x] I have reviewed my code
   - [x] I have commented my code, particularly in hard-to-understand areas
   - [ ] I have fully tested my code

### CODE REVIEW

**Review by CrowRising — CHANGES_REQUESTED** [review]
I don't think we need the last two methods in the match poro...lines 36 thru 51
We should not be manipulating the data in that way on the back end

File: app/poros/match.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+
+  def initialize(data)
+    # @access_token = access_token
+    @probability = format_probability(data[:probability])
+
+    @psychoactive = data[:details][:psychoactive]
+
+    @api_id = data[:id] 
+    @common_name = data[:details][:common_names].first
```
Comment (JCBeldo) [inline app/poros/match.rb:23]: I think they might want multiple names

File: app/services/kindwise_service.rb [hunk 2]
```diff
@@ -9,24 +9,14 @@ def conn
...
-      req.params["details"] = [
-        'common_names',
-        'edibility',
-        'description',
-        'psychoactive',
-        'taxonomy',
-        'characteristic',
-        'look_alike'
-      ].join(',')
+      req.params['details'] = 'common_names,description,edibility,image,taxonomy,psychoactive,characteristic,look_alike,description'
```
Comment (JCBeldo) [inline app/services/kindwise_service.rb:18]: I like this all grouped together

**Comment by MWMJohnson** [comment]
Good Callout Crow, we will let FE know to manipulate on their end.

File: spec/graphql/resolvers/users/users_resolver_spec.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
+
+RSpec.describe Resolvers::Users::UsersResolver do
+  describe '#resolve' do
+    it 'fetches all users' do
+      user_1 = User.create!(name: "Billy D.", email: "billyd@email.com")
+      user_2 = User.create!(name: "Sally D.", email: "sallyd@email.com")
+      user_3 = User.create!(name: "Sunny D.", email: "sunnyd@email.com")
```
Comment (JCBeldo) [inline spec/graphql/resolvers/users/users_resolver_spec.rb:8]: purplestuff@email.com

**Review by JCBeldo — CHANGES_REQUESTED** [review]
Should we remove that .first for common names?

**Review by JCBeldo — APPROVED** [review]

**Review by CrowRising — APPROVED** [review]

**Pull request merged**